ankh/cli.py

```diff
@@ -117,46 +178,138 @@ def template_ingress_hosts(global_config):
+            return current_context_command(args)
+        if command == 'get-contexts':
+            return get_contexts_command(args)
+        if command == 'set-context':
+            if len(args.command) != 2:
+                logger.error("Set context command requires exactly 1 postional arguments after 'set-context'")
+                sys.exit(1)
+            return set_context_command(args, args.command[1])
+        if command == 'help':
+            parser.print_help()
+            return 0
+    except KeyboardInterrupt:
+        logger.info("Interrupted")
+        sys.exit(1)
 
-    logger.info("Running from directory %s" % os.getcwd())
-    for config_file in args.config_files:
-        if os.path.exists(config_file):
-            logger.info("Using config file " + config_file)
-            with open(config_file, 'r') as f:
-                config = yaml.load(f)
-                merge_config(global_config, config)
-        elif config_file == 'deploy.yaml':
-            logger.info("Ignoring missing default config file " + config_file)
+    # We need the first pass of config to determine dependencies.
+    logger.info("Gathering global configuration...")
+    global_config = gather_config(args, log=True)
+
+    if command == 'config':
+        yaml.dump(global_config, sys.stdout, default_flow_style=False)
+        return 0
+
+    dependencies = []
+    if 'admin_dependencies' in global_config:
+        if global_config.get('cluster_admin', False):
+            logger.debug("Current context has cluster_admin: adding admin_dependencies")
+            dependencies.extend(global_config['admin_dependencies'])
         else:
-            logger.error("Cannot find config file " + config_file)
-            sys.exit(1)
+            logger.debug("Current context does not have cluster_admin: skipping admin_dependencies")
+    if 'dependencies' in global_config:
+        dependencies.extend(global_config['dependencies'])
+
+    if len(dependencies) == 0:
+        # common-case: only run in this directory
+        return run('.', args)
+
+    # Satisfy each dependency by changing into that directory and running.
+    # Recursive dependencies are prevented in run().
+    logger.debug("Found dependencies: %s" % str(dependencies))
+    for dep in dependencies:
+        logger.info("Satisfying dependency: %s" % dep)
+        old_dir = os.getcwd()
+        os.chdir(dep)
+        r = run(dep, args)
+        os.chdir(old_dir)
+        if r != 0:
+            return r
+    return 0
+
+def run(base_dir, args):
+    logger.info("Running from directory %s" % os.getcwd())
+
+    logger.info("Gathering local configuration...")
+    global_config = gather_config(args, log=True)
+    if 'dependencies' in global_config:
+        logger.error("Base directory %s contains dependencies in its config, but we're processing dependencies right now. Recursive dependencies aren't supported, so eliminate them before proceeding." % os.getcwd())
```

Comment:
Might mention something about being in the global config to make the error messaging a little clearer for users

Reply:
good call.

Reply:
done